### ISSUE TITLE
Fix imports after refactoring

### ISSUE THREAD

**Issue by e-vissch** [microsoft/pylance-release#3929]
Pycharm has a feature which automatically fixes imports on refactoring/renaming a file. From memory it asks the user whether to search for usages, if yes it gets the user to go through and check how to fix imports or press a button to accept automatic fixes. I personally think this would be a useful addition to vscode.

I saw a request for this back in 2018 that was closed [#1808](https://github.com/Microsoft/vscode-python/issues/1808 ). Is there any scope to revisit this?

**Comment by rchiodo** [comment]
@heejaechang is your new move code handling this case already? Or would this require additional work?

**Comment by heejaechang** [comment]
it is already supported through LSP rename file capability. if it didn't work for you, can you provide a repro steps so we can investigate why it didn't work for you?

**Comment by e-vissch** [comment]
Ok so I have just tested it, and yes it sometimes works, although it sometimes does not work.

It seems to work if I first rename a file. If I try to rename a file that I have recently renamed it often does not work. 
I.e if I: 
1. rename a file from file.py to file_test.py - it usually works (as in comes up with pop up box confirming refactor by python extension, and usages in other files updated) 
2. if I then rename file_test.py back to file.py it does not come up with the box and renames the file but not any of the usages. 

Interestingly if I repeat the two steps it seems to work on step 2. of the repeated process. 

Also it doesn't refactor imports in ipython files, but maybe that's expected? 

I can't remember the circumstances of when I first opened this issue if it was a case of the above or something different.

**Comment by e-vissch** [comment]
Okie, I'm not entirely sure what you mean, but had a look at other issues for inspiration.

## Environment Data:
- Language Server version: Pylance  v2023.2.20 
- Vscode Python  v2023.2.0 
- OS and version: macos 12.5

```Version: 1.75.1
Commit: 441438abd1ac652551dbe4d408dfcec8a499b8bf
Date: 2023-02-08T21:34:01.965Z
Electron: 19.1.9
Chromium: 102.0.5005.194
Node.js: 16.14.2
V8: 10.2.154.23-electron.0
OS: Darwin arm64 21.6.0
Sandboxed: No
```

## Code Snippet:

Contents of `file.py`:
```python
def test_function():
    return
```
Contents of `file_2.py`:
```python
from file import test_function
```

## Repro steps:

1. Have `file.py` open but NOT `file_2.py` open
2. Rename `file.py` to `file_3.py` -> accept python refactoring.
3. Rename `file_3.py` to `file.py`

### Expected Behaviour
After step 2. of above `file_2.py` now says:
```python
from file_3 import test_function
```
After step 3, I would expect to be asked if I want to refactor and if yes, `file_2.py` would say:
```python
from file import test_function
```

## Actual Behaviour
Not asked to refactor and `file_3.py` is renamed to `file.py` without updating `file_2.py`, so `file_2.py` continues to say:
```python
from file_3 import test_function
```
But now with a yellow squiggly line as `file_3.py` no longer exists. 

### Logs
N/A

**Comment by heejaechang** [comment]
does the issue happen for any rename after the very first rename? or only if we rename back to the original file name?

**Comment by e-vissch** [comment]
only if it's back to the original

**Comment by debonte** [comment]
Looks like this was actually fixed in 2023.2.31 and was missing the `fixed in next version` label so it wasn't announced.

**Comment by thoughtfuldata** [comment]
I'm getting this issue on:

Version: 1.87.2 (user setup)
Commit: 863d2581ecda6849923a2118d93a088b0745d9d6
Date: 2024-03-08T15:20:17.278Z
Electron: 27.3.2
ElectronBuildId: 26836302
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Windows_NT x64 10.0.22631

Maybe a regression?

**Comment by rchiodo** [comment]
@thoughtfuldata it would be best if you opened a new issue. The original poster's repo does not reproduce for me with the latest Pylance bits. You might also check your Pylance version.

**Comment by thoughtfuldata** [comment]
@rchiodo Alright,  I will make a new one, its the same repro steps taken in #4964. I noticed someone else had the issue as well.

My pylance version is `v2024.4.1`
